nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -889,4 +891,283 @@ def write_recording(
+        warnings.warn("The nwbfile already contains units. These units will not be over-written.")
+
+
+def write_sorting(
+    sorting: se.SortingExtractor,
+    save_path: PathType = None,
+    overwrite: bool = False,
+    nwbfile=None,
+    property_descriptions: Optional[dict] = None,
+    skip_properties: Optional[List[str]] = None,
+    skip_features: Optional[List[str]] = None,
+    use_times: bool = True,
+    **nwbfile_kwargs
+):
+    """
+    Primary method for writing a SortingExtractor object to an NWBFile.
+
+    Parameters
+    ----------
+    sorting: SortingExtractor
+    save_path: PathType
+        Required if an nwbfile is not passed. The location where the NWBFile either exists, or will be written.
+    overwrite: bool
+        If using save_path, whether or not to overwrite the NWBFile if it already exists.
+    nwbfile: NWBFile
+        Required if a save_path is not specified. If passed, this function
+        will fill the relevant fields within the nwbfile. E.g., calling
+        spikeextractors.NwbRecordingExtractor.write_recording(
+            my_recording_extractor, my_nwbfile
+        )
+        will result in the appropriate changes to the my_nwbfile object.
+    property_descriptions: dict
+        For each key in this dictionary which matches the name of a unit
+        property in sorting, adds the value as a description to that
+        custom unit column.
+    skip_properties: list of str
+        Each string in this list that matches a unit property will not be written to the NWBFile.
+    skip_features: list of str
+        Each string in this list that matches a spike feature will not be written to the NWBFile.
+    use_times: bool (optional, defaults to False)
+        If True, the times are saved to the nwb file using sorting.frame_to_time(). If False (defualut),
+        the sampling rate is used.
+    nwbfile_kwargs: dict
+        Information for constructing the nwb file (optional).
+        Only used if no nwbfile exists at the save_path, and no nwbfile
+        was directly passed.
+    """
+    assert save_path is None or nwbfile is None, \
+        "Either pass a save_path location, or nwbfile object, but not both!"
+    if nwbfile is not None:
+        assert isinstance(nwbfile, pynwb.NWBFile), "'nwbfile' should be a pynwb.NWBFile object!"
+
+    if nwbfile is None:
+        if Path(save_path).is_file() and not overwrite:
+            read_mode = 'r+'
+        else:
+            read_mode = 'w'
+
+        with pynwb.NWBHDF5IO(str(save_path), mode=read_mode) as io:
+            if read_mode == 'r+':
+                nwbfile = io.read()
+            else:
+                default_nwbfile_kwargs = dict(
+                    session_description="Auto-generated by NwbSortingExtractor without description.",
+                    identifier=str(uuid.uuid4()),
+                    session_start_time=datetime(1970, 1, 1)
+                )
+                default_nwbfile_kwargs.update(**nwbfile_kwargs)
```

Comment:
It definitely did. TBH this aspect of the functionality is rarely if ever used, but yes in our nwb-conversion-tools setup it would pertain to metadata. Note that in our own ecosystem this functionality would only actually occur for one-line conversions of a single Recording/SortingExtractor object.